B/Birch/bundled/glob_nomagic.patch

```diff
@@ -0,0 +1,11 @@
+--- driver/src/build/misc.cpp.orig
++++ driver/src/build/misc.cpp
+@@ -36,7 +36,7 @@
+ std::list<fs::path> birch::glob(const std::string& pattern) {
+   std::list<fs::path> results;
+   glob_t matches;
+-  glob(pattern.c_str(), GLOB_NOMAGIC, 0, &matches);
++  glob(pattern.c_str(), 0, 0, &matches);
```

Comment:
I am not sure if this change causes any problems when building other Birch projects apart from the standard library. Therefore so far this patch is only applied when the driver is built for the host system on `*-musl` but not for building Birch for `*-musl` target systems (these are disabled).

According to the [libc docs](https://www.gnu.org/software/libc/manual/html_node/More-Flags-for-Globbing.html), the flag `GLOB_NOMAGIC` provides the following functionality:

> If the pattern contains no wildcard constructs (it is a literal file name), return it as the sole “matching” word, even if no file exists by that name.

Naively, it seemed safe to remove the flag since it only leads to a different behaviour if no file of this name exists. Currently, in this case users are presented with a warning 
(https://github.com/lawmurray/Birch/blob/0f1461151837d3012b3aedc8121955752a58995e/driver/src/build/Driver.cpp#L1252-L1254), and with this change the path would be dropped silently instead.

I guess @lawmurray knows if this change causes any other problems, or if it can be applied safely in general (and hence would allow to build Birch for more targets).

Reply:
This will be fine, preferable even as it will help catch errors. Can we push these and other changes upstream to the Birch codebase?

Reply:
Sure, I'll open some PRs.